src/main/java/com/iexec/commons/poco/encoding/AssetDataEncoder.java

```diff
@@ -0,0 +1,126 @@
+
+    /**
+     * Encodes data for {@code createApp} transaction.
+     *
+     * @param owner     Owner Ethereum address
+     * @param name      App name
+     * @param type      App type, unique supported value should be DOCKER
+     * @param multiaddr Docker registry download address
+     * @param checksum  Docker image SAH256 checksum
+     * @param mrenclave TEE configuration if applicable, should be a valid JSON string of {@link com.iexec.commons.poco.tee.TeeEnclaveConfiguration}
+     * @return encoded data
+     */
+    public static String encodeApp(String owner, String name, String type, String multiaddr, String checksum, String mrenclave) {
+        long offset = 6;
+        StringBuilder sb = new StringBuilder(CREATE_APP_SELECTOR);
+        sb.append(toHexString(owner));
+
+        sb.append(toHexString(BigInteger.valueOf(offset * 32)));
+        String nameContrib = TypeEncoder.encode(new DynamicBytes(name.getBytes(StandardCharsets.UTF_8)));
+
+        offset += nameContrib.length() / 64;
+        sb.append(toHexString(BigInteger.valueOf(offset * 32)));
+        String typeContrib = TypeEncoder.encode(new DynamicBytes(type.getBytes(StandardCharsets.UTF_8)));
+
+        offset += typeContrib.length() / 64;
+        sb.append(toHexString(BigInteger.valueOf(offset * 32)));
+        String multiaddrContrib = TypeEncoder.encode(new DynamicBytes(multiaddr.getBytes(StandardCharsets.UTF_8)));
+
+        sb.append(checksum);
+
+        offset += multiaddrContrib.length() / 64;
+        sb.append(toHexString(BigInteger.valueOf(offset * 32)));
+        String mrenclaveContrib = TypeEncoder.encode(new DynamicBytes(mrenclave.getBytes(StandardCharsets.UTF_8)));
+
+        sb.append(nameContrib);
+        sb.append(typeContrib);
+        sb.append(multiaddrContrib);
+        sb.append(mrenclaveContrib);
+        log.debug("app tx [data:{}]", sb);
+        return sb.toString();
+    }
+
+    /**
+     * Encodes data for {@code createDataset} transaction.
+     *
+     * @param owner     Owner Ethereum address
+     * @param name      Dataset name, can be empty
+     * @param multiaddr Download address, can be a http link or an IPFS multiaddr
+     * @param checksum  SHA256 checksum of the dataset
+     * @return encoded data
+     */
+    public static String encodeDataset(String owner, String name, String multiaddr, String checksum) {
+        long offset = 4;
+        StringBuilder sb = new StringBuilder(CREATE_DATASET_SELECTOR);
+
+        String datasetNameOffset = toHexString(BigInteger.valueOf(offset * 32));
+        String datasetNameContrib = TypeEncoder.encode(new DynamicBytes(name.getBytes(StandardCharsets.UTF_8)));
+        offset += datasetNameContrib.length() / 64;
+
+        String datasetAddrOffset = toHexString(BigInteger.valueOf(offset * 32));
+        String datasetAddrContrib = TypeEncoder.encode(new DynamicBytes(multiaddr.getBytes(StandardCharsets.UTF_8)));
```

Comment:
As we don't prefix app's vars with `app`, should we not prefix dataset's vars with `dataset`?
```suggestion
        String nameOffset = toHexString(BigInteger.valueOf(offset * 32));
        String nameContrib = TypeEncoder.encode(new DynamicBytes(name.getBytes(StandardCharsets.UTF_8)));
        offset += datasetNameContrib.length() / 64;

        String addrOffset = toHexString(BigInteger.valueOf(offset * 32));
        String addrContrib = TypeEncoder.encode(new DynamicBytes(multiaddr.getBytes(StandardCharsets.UTF_8)));
```